app/models/stage.rb

```diff
@@ -154,6 +156,14 @@ def send_github_notifications?
     update_github_pull_requests
   end
 
+  def production?
+    if ENV['DEPLOY_GROUP_FEATURE']
+      deploy_groups.empty? ? super : deploy_groups.any? { |deploy_group| deploy_group.environment.is_production? }
```

Comment:
True, though I think I'll leave that for another PR for the moment. I really want to get this merged, and for now we don't need the query, as the next step is for environment dashboards which use the association from the other direction.